src/HttpClientPool/HttpClientPoolItem.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Common\HttpClientPool;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * A HttpClientPoolItem represent a HttpClient inside a Pool.
+ *
+ * It is disabled when a request failed and can be reenabled after a certain number of seconds.
+ * It also keep tracks of the current number of open requests the client is currently being sending
+ * (only usable for async method).
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface HttpClientPoolItem extends HttpClient, HttpAsyncClient
```

Comment:
Not a fan of that, creating an interface because of mocking makes me sad, having an interface add a new dependency and i would not have that here, HttpClientPoolItem is an internal thing, so we should just tag it `@internal` or `@final` IMO

Reply:
yeah, you are right with that one. removed the interface and documented that this is internal only.